src/builder.rs

```diff
@@ -160,180 +156,13 @@ impl AstarteBuilder {
-        cn: &str,
-    ) -> Result<(), AstarteBuilderError> {
-        let ifaces = self
-            .interfaces
-            .clone()
-            .into_iter()
-            .filter(|i| i.1.get_ownership() == Ownership::Server);
-
-        client
-            .subscribe(
-                cn.to_owned() + "/control/consumer/properties",
-                rumqttc::QoS::ExactlyOnce,
-            )
-            .await?;
-
-        for i in ifaces {
-            client
-                .subscribe(
-                    cn.to_owned() + "/" + i.1.name() + "/#",
-                    rumqttc::QoS::ExactlyOnce,
-                )
-                .await?;
-        }
-
-        Ok(())
-    }
-
-    /// build Astarte client, call this before `connect`
-    pub async fn build(&mut self) -> Result<(), AstarteBuilderError> {
-        let cn = format!("{}/{}", self.realm, self.device_id);
-
-        if self.interfaces.is_empty() {
-            return Err(AstarteBuilderError::MissingInterfaces);
-        }
-
-        let Bundle(pkey_bytes, csr_bytes) = Bundle::new(&cn)?;
-
-        let private_key = pemfile::pkcs8_private_keys(&mut pkey_bytes.as_slice())
-            .map_err(|_| AstarteBuilderError::ConfigError("failed pkcs8 key extraction".into()))?
-            .remove(0);
-
-        let csr = String::from_utf8(csr_bytes)
-            .map_err(|_| AstarteBuilderError::ConfigError("bad csr bytes format".into()))?;
-
-        let certificate_pem = self.populate_credentials(&csr).await?;
-
-        let broker_url = self.populate_broker_url().await?;
-
-        let mqtt_opts = self.build_mqtt_opts(&certificate_pem, &broker_url, &private_key)?;
-
-        self.build_options = Some(BuildOptions {
-            private_key,
-            csr,
-            certificate_pem,
-            broker_url,
-            mqtt_opts,
-        });
-
-        Ok(())
-    }
-
-    /// Creates and connects an Astarte client
-    pub async fn connect(&mut self) -> Result<AstarteSdk, AstarteBuilderError> {
-        let cn = format!("{}/{}", self.realm, self.device_id);
-
-        let build_options = self
-            .build_options
-            .clone()
-            .ok_or_else(|| AstarteBuilderError::ConfigError("Missing or failed build".into()))?;
-
-        // TODO: make cap configurable
-        let (client, eventloop) = AsyncClient::new(build_options.mqtt_opts.clone(), 50);
-
-        self.subscribe(&client, &cn).await?;
-
-        let device = AstarteSdk {
-            realm: self.realm.to_owned(),
-            device_id: self.device_id.to_owned(),
-            credentials_secret: self.credentials_secret.to_owned(),
-            pairing_url: self.pairing_url.to_owned(),
-            build_options,
-            client,
-            eventloop: Arc::new(tokio::sync::Mutex::new(eventloop)),
-            interfaces: Interfaces::new(self.interfaces.clone()),
-            database: self.database.clone(),
-        };
-
-        Ok(device)
+    pub fn build(&self) -> Self {
```

Comment:
The idea of "building" options might be confusing, we should come up with better naming

Reply:
I am open to proposals

get()? make()?

Reply:
IMHO **build method** should be removed. @rbino what do you think?

Reply:
After some discussion, `build` was approved since that's actually a builder for the `AstarteOptions` now